doc/mgmt/SONiC_TIMEZONE_HLD.md

```diff
@@ -0,0 +1,546 @@
+- SONiC clock model in sonic-system-clock.yang to get timezone from timedatectl output, including UTC offset     
+       
+**time and date**   
+     
+Details to be added.      
+      
+### 1.2.3 SAI Overview
+
+# 2 Functionality
+
+## 2.1 Target Deployment Use Cases
+Manage/configure timezone via gNMI, REST and CLI interfaces.    
+Manage/configure time&date via gNMI, REST and CLI interfaces.    
+           
+## 2.2 Functional Description
+Provide CLI, gNMI and REST supports for NTP related configurations.
+            
+## 2.3 Backend change to support new configurations
+**timezone**       
+     
+- transformer functions to    
+   * set "system" as key for CLOCK table in CONFIG DB   
+   * rpc function to get output of timedatectl for OpenConfig YANG    
+   * rpc function to get output of timedatectl for SONiC YANG      
+   * subscribe to listen for CLOCK change in CONFIG DB    
+   * set local timezone for glog   
+- customer cvl validation function to reject timezone if the timezone name does not exist under /usr/share/zoneinfo. This is for the case when timezone configuration is done from SONiC YANG.     
+- glog patch to handle local timezone change for the timestamp    
+- SONiC click CLI enhancement if possible.         
+    
+**time and date**   
+     
+Details to be added.     
+    
+# 3 Design
+    
+## 3.1 Overview    
+Suppport timezone and time&date configuration in Management Framework.    
+    
+## 3.2 DB Changes
+
+### 3.2.1 CONFIG DB
+This feature will allow users to make CLOCK change in CONFIG DB, and to get CLOCK configuration.
+
+```
+   "CLOCK": {
+        "system": {
+            "timezone_name": "US/Hawaii"
+        }
+    }
+```
+    
+### 3.2.2 APP DB
+
+### 3.2.3 STATE DB
+
+### 3.2.4 ASIC DB
+
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+
+### 3.3.1 Orchestration Agent
+
+### 3.3.2 Other Process
+
+## 3.4 SyncD
+
+## 3.5 SAI
+
+## 3.6 User Interface
+
+### 3.6.1 Data Models
+
+YANG models needed for timezone handling in the management framework:
+1. **openconfig-system.yang**    
+    
+3. **sonic-system-clock.yang**    
+     
+Supported yang objects and attributes:    
+    
+```diff
++ module: openconfig-system.yang
+
++     +--rw system
+      ...
++         +--rw clock
++         |  +--rw config
++         |  |  +--rw timezone-name?   timezone-name-type
++         |  +--ro state
++         |     +--ro timezone-name?   timezone-name-type
+```
+
+```diff
++  module: openconfig-system-private.yang
+
++  rpcs:
++    +---x get-timezone
++       +--ro output
++          +--ro utc-offset?      string
++          +--ro timezone-abbr?   string
+```
+
+```diff
++    module: sonic-system-clock
++      +--rw sonic-system-clock
++         +--rw CLOCK
++            +--rw CLOCK_LIST* [systemclock_key]
++               +--rw systemclock_key    enumeration
++               +--rw timezone_name?     string
+
++  rpcs:
++    +---x get_timezone

```

Comment:
We don't need this RPC in SONiC YANG as we have it already in OC-YANG.

Reply:
The OC-YANG only has timzone name, while the "show clock timezone" will include UTC offset as well.